File: app/src/main/java/co/kr/sumai/spinner/HintSpinner.java
package co.kr.sumai.spinner;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.List;

public class HintSpinner<T> extends ArrayAdapter {
    public HintSpinner(Context content, int resource, T[] objects) {
        super(content, resource, objects);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        View v = super.getView(position, convertView, parent);


        if(position == getCount()) {
            ((TextView) v.findViewById(android.R.id.text1)).setText("");
            ((TextView) v.findViewById(android.R.id.text1)).setHint((CharSequence) getItem(getCount()));
        }

        return v;
    }

    @Override
    public int getCount() {
        return super.getCount() - 1;
    }
}

File: app/src/main/java/co/kr/sumai/spinner/HintSpinner.kt
package co.kr.sumai.spinner

import android.R
import android.content.Context
import android.graphics.Color
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

class HintSpinner<T>(content: Context?, resource: Int, objects: Array<T>?) : ArrayAdapter<Any?>(content!!, resource, objects!!) {
    override fun isEnabled(position: Int): Boolean {
        return position > 0
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        val v: TextView = super.getDropDownView(position, convertView, parent) as TextView
        v.setTextColor(if(position > 0) Color.BLACK else Color.GRAY)
        return v
    }
}